pandas/tests/groupby/test_groupby.py

```diff
@@ -1994,3 +1994,22 @@ def test_dup_labels_output_shape(groupby_func, idx):
 
     assert result.shape == (1, 2)
     tm.assert_index_equal(result.columns, idx)
+
+
+def test_groupby_crash_on_nunique(axis):
+    # Fix following 30253
+    df = pd.DataFrame(
+        {("A", "B"): [1, 2], ("A", "C"): [1, 3], ("D", "B"): [0, 0]},
+        index=pd.bdate_range("20191212", "20191213"),
+    )
+    df.columns.names = ["COL1", "COL2"]
+
+    axis_number = df._get_axis_number(axis)
+    if not axis_number:
+        df = df.T
+
+    nunique_t = df.T.groupby(level=0, axis=axis).nunique().T
```

Comment:
Can you try and simplify this test a bit? Somewhat hard to follow with all of the transposing going on. Would be preferable if you can construct with literal values

Reply:
I will rework the test, with having a literal expected value and asserting the groupby is the expected literal